File: Cs-Connect-java/app/src/main/java/com/csestateconnect/www/csconnect/adapters/ProjectWowFactorAdapter.java
package com.csestateconnect.www.csconnect.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.csestateconnect.www.csconnect.ProjectDetailActivity;
import com.csestateconnect.www.csconnect.R;
import com.csestateconnect.www.csconnect.models.project.Project;
import com.csestateconnect.www.csconnect.models.project.ProjectWowFactor;

import java.util.LinkedList;
import java.util.List;

public class ProjectWowFactorAdapter extends RecyclerView.Adapter<ProjectWowFactorAdapter.WowViewHolder> {

    private final List<ProjectWowFactor> mWowFactorList;
    private LayoutInflater mInflater;
    private Context mContext;
   int placeholder = -1;

    public ProjectWowFactorAdapter(Context context, List<ProjectWowFactor> mProjectList) {
        mInflater = LayoutInflater.from(context);
        this.mWowFactorList = mProjectList;
        this.mContext = context;
    }

    class WowViewHolder extends RecyclerView.ViewHolder {
        public final TextView wowFactorsNameView;

        final ProjectWowFactorAdapter mAdapter;

        public WowViewHolder(View itemView, ProjectWowFactorAdapter adapter) {
            super(itemView);
            wowFactorsNameView = itemView.findViewById(R.id.wow_factors_txt);

            this.mAdapter = adapter;
        }

    }

    @NonNull
    @Override
    public WowViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View mItemView = mInflater.inflate(R.layout.wow_factor_list, parent, false);
        return new WowViewHolder(mItemView, this);
    }

    @Override
    public void onBindViewHolder(@NonNull WowViewHolder projectViewHolder, final int position) {
        ProjectWowFactor mCurrentProject = mWowFactorList.get(position);

        projectViewHolder.wowFactorsNameView.setText(mCurrentProject.getName());

    }

    @Override
    public int getItemCount() {
        return mWowFactorList.size();
    }
}

File: CsConnect/app/src/main/java/com/csestateconnect/adapters/projectAdapters/ProjectWowFactorAdapter.kt
package com.csestateconnect.adapters.projectAdapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.csestateconnect.R
import com.csestateconnect.db.data.projects.ProjectWowFactor
import java.lang.Exception


class ProjectWowFactorAdapter(val wowFactorList: List<ProjectWowFactor?>?) : RecyclerView.Adapter<ProjectWowFactorAdapter.ViewHolder>() {

    var count: Int = 0
    override fun onCreateViewHolder(p0: ViewGroup, viewType: Int): ViewHolder {
        val v: View = LayoutInflater.from(p0.context)
            .inflate(com.csestateconnect.R.layout.project_wow_factor_item,p0,false)

        return ViewHolder(v)
    }

/*class CustomViewHolder(val view:View):RecyclerView.ViewHolder(view)*/
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

    try {
        holder.project_wowFactorTxt.text = wowFactorList!!.get(position)!!.name

    }catch (e: Exception){
        e.printStackTrace()
    }


    }
    override fun getItemCount(): Int {
        var count: Int =0
      try {
          count  = wowFactorList!!.size
      }catch (e: Exception){
          e.printStackTrace()
      }
        return count
    }


    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        val project_wowFactorTxt: TextView = itemView.findViewById(R.id.wow_factors_txt)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }
}